ios/brave-ios/Sources/Brave/Frontend/Browser/Tabs/TabGrid/TabGridViewModel.swift

```diff
@@ -113,20 +113,19 @@ class TabGridViewModel {
 
   func closeSelectedTab() {
     guard let tab = tabManager.selectedTab else { return }
-    withAnimation {
-      tabManager.removeTab(tab)
-    }
+    closeTab(tab)
   }
 
   func closeTab(_ tab: any TabState) {
     withAnimation {
       tabManager.removeTab(tab)
+      tabManager.addTabToRecentlyClosed(tab)
     }
   }
 
   func closeAllTabs() {
     withAnimation {
-      tabManager.removeAllTabsForPrivateMode(isPrivate: isPrivateBrowsing)
+      tabManager.removeTabsWithUndoToast(tabManager.tabsForCurrentMode)
```

Comment:
@jamesmudgett @kleantzogu should bulk removals be added to the recently closed list? The old UI did not have any user facing button for closing all tabs, only a keyboard shortcut and that keyboard shortcut did not add every tab to the recently closed tab list, only showed an undo toast.

Options:
- Show the undo toast, don't add anything to recently closed list
- Add every tab to the recently closed list, dont bother with an undo list
- Do both, but when the user undoes, we may have to go purge the recently closed list items that were just added

Reply:
I guess a fourth option: Add a confirmation dialog stating that removing all tabs cant be undone and then don't do either (undo toast/add to recently closed list)